Integrations/SMIME_Messaging/SMIME_Messaging.py

```diff
@@ -0,0 +1,198 @@
+    headers = signed_message[0].replace(': ', '=').replace('\n', ',')
+    body = signed_message[2:]
+    context = {
+        'SMIME': {
+            'Message': body,
+            'Headers': headers
+        }
+    }
+
+    return signed, context
+
+
+def encrypt_email_body(client: Client, args: Dict):
+    """ generate an S/MIME-encrypted message
+
+    Args:
+        client: Client
+        args: Dict
+
+    """
+    message_body = args.get('message', '').encode('utf-8')
+
+    # Make a MemoryBuffer of the message.
+    buf = makebuf(message_body)
+
+    # Load target cert to encrypt to.
+    x509 = X509.load_cert(client.public_key_file)
+    sk = X509.X509_Stack()
+    sk.push(x509)
+    client.smime.set_x509_stack(sk)
+
+    # Set cipher: 3-key triple-DES in CBC mode.
+    client.smime.set_cipher(SMIME.Cipher('des_ede3_cbc'))
+
+    # Encrypt the buffer.
+    p7 = client.smime.encrypt(buf)
+
+    # Output p7 in mail-friendly format.
+    out = BIO.MemoryBuffer()
+
+    client.smime.write(out, p7)
+    encrypted_message = out.read().decode('utf-8')
+    message = encrypted_message.split('\n\n')
+    headers = message[0]
+    new_headers = headers.replace(': ', '=').replace('\n', ',')
+
+    entry_context = {
+        'SMIME': {
+            'Message': encrypted_message,
+            'Headers': new_headers
+        }
+    }
+    return encrypted_message, entry_context
+
+
+def verify(client: Client, args: Dict):
+    """ Verify the signature
+
+    Args:
+        client: Client
+        args: Dict
+
+    """
+    signed_message = demisto.getFilePath(args.get('signed_message'))
+
+    # Load the signer's cert.
+    x509 = X509.load_cert(client.public_key_file)
+    sk = X509.X509_Stack()
+    sk.push(x509)
+    client.smime.set_x509_stack(sk)
+
+    # Load the signer's CA cert. In this case, because the signer's
+    # cert is self-signed, it is the signer's cert itself.
+    st = X509.X509_Store()
+    st.load_info(client.public_key_file)
+    client.smime.set_x509_store(st)
+
+    # Load the data, verify it.
+    p7, data = SMIME.smime_load_pkcs7(signed_message['path'])
+    v = client.smime.verify(p7, data, flags=SMIME.PKCS7_NOVERIFY)
+
+    human_readable = f'The signature verified\n\n{v}\n{data}'
+    return human_readable, {}
+
+
+def decrypt_email_body(client: Client, args: Dict):
+    """ Decrypt the message
+
+    Args:
+        client: Client
+        args: Dict
+    """
+
+    encrypt_message = demisto.getFilePath(args.get('encrypt_message'))
+    # Load private key and cert.
+    client.smime.load_key(client.private_key_file, client.public_key_file)
+
+    # Load the encrypted data.
+    p7, data = SMIME.smime_load_pkcs7(encrypt_message['path'])
+
+    # Decrypt p7.
+    out = client.smime.decrypt(p7).decode('utf-8')
+    entry_context = {
+        'SMIME': {
+            'Message': out,
+        }
+    }
+    human_readable = f'The encrypted message is: \n{out}'
+
+    return human_readable, entry_context
+
+
+def test_module():
```

Comment:
can we do something in the test? maybe sign?

Reply:
I called in the test_module the encrypt and decrypt so we can check that the keys are correct. 
Also added a call to this function in the unittest